language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Exception.hs

```diff
@@ -29,6 +29,8 @@ data MachineError err
       -- ^ An attempt to evaluate an open term.
     | ConstAppMachineError ConstAppError
       -- ^ An attempt to compute a constant application resulted in 'ConstAppError'.
+    | OutOfExError
+      -- ^ An attempt to continue evaluation without sufficient Ex
```

Comment:
We talked about that, this is not an exceptional situation, it's the "return `EvaluationFailure`" situation. @michaelpj right?

Reply:
Yes, I agree. Running out of memory (like computing to `error`) is a "normal" exit condition for the machine. These exceptions are for abnormal exit conditions e.g. type-incorrect input.

Reply:
(This is somewhat vague, obviously.)

Reply:
Although I can see that's awkward for the way you've structured this. We could catch the out-of-ex error and put it into `EvaluationFailure`?

Reply:
Although I can see that's awkward for the way you've structured this. We could catch the out-of-ex error and put it into `EvaluationFailure`?

Reply:
I don't think we should throw that error just to immediately catch it.